examples/01-filter/extract-surface.py

```diff
@@ -0,0 +1,71 @@
+"""
+Extract Surface
+~~~~~~~~~~~~~~~
+
+You can extract the surface of nearly any object within ``pyvista``
+using the ``extract_surface`` filter.
+"""
+
+# sphinx_gallery_thumbnail_number = 2
+
+import numpy as np
+import pyvista as pv
+from vtk import VTK_QUADRATIC_HEXAHEDRON
+
+###############################################################################
+# Create a quadratic cell and extract its surface
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Here we create a single hexahedral cell and then extract its surface
+# to demonstrate how to extract the surface of a UnstructuredGrid.
+
+
+lin_pts = np.array([[-1, -1, -1],  # point 0
+                    [ 1, -1, -1],  # point 1
+                    [ 1,  1, -1],  # point 2
+                    [-1,  1, -1],  # point 3
+                    [-1, -1,  1],  # point 4
+                    [ 1, -1,  1],  # point 5
+                    [ 1,  1,  1],  # point 6
+                    [-1,  1,  1]], np.double)  # point 7
+
+# these are the "midside" points of a quad cell
+quad_pts = np.array([
+    (lin_pts[1] + lin_pts[0])/2.0,
+    (lin_pts[1] + lin_pts[2])/2.0,
+    (lin_pts[2] + lin_pts[3])/2.0,
+    (lin_pts[3] + lin_pts[0])/2.0,
+    (lin_pts[4] + lin_pts[5])/2.0,
+    (lin_pts[5] + lin_pts[6])/2.0,
+    (lin_pts[6] + lin_pts[7])/2.0,
+    (lin_pts[7] + lin_pts[4])/2.0,
+    (lin_pts[0] + lin_pts[4])/2.0,
+    (lin_pts[1] + lin_pts[5])/2.0,
+    (lin_pts[2] + lin_pts[6])/2.0,
+    (lin_pts[3] + lin_pts[7])/2.0], np.double)
+
+# introduce a minor variation to the location of the mid-side points
+quad_pts += np.random.random(quad_pts.shape)*0.3
+pts = np.vstack((lin_pts, quad_pts))
+
+# create the grid
+
+# If you are using vtk>=9, you do not need the offset array
+offset = np.array([0])
+cells = np.asarray(np.hstack((20, np.arange(20))), dtype=np.int64)
+celltypes = np.array([VTK_QUADRATIC_HEXAHEDRON])
+grid = pv.UnstructuredGrid(offset, cells, celltypes, pts)
+
+# finally, extract the surface and plot it
+surf = grid.extract_surface()
+surf.plot(show_scalar_bar=False)
+
+
+###############################################################################
+# Quadratic Surface Subdivision
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Should your UnstructuredGrid contain quadratic cells, you can
+# generate a smooth surface based on the "shape functions" of each
+# cell.
```

Comment:
```suggestion
# generate a smooth surface based on the position of the 
"mid-edge" nodes.  This allows the plotting of cells
containing curvature.  For additional reference, please see:
https://prod.sandia.gov/techlib-noauth/access-control.cgi/2004/041617.pdf
```

Reply:
I'm missing context here for sure.  It's too much to explain in a basic example, but I can give a two sentence explanation and a reference for those who want to read more.  For your edification, "quadratic elements" are widely used in finite element analysis to compute better approximations of static/harmonic characteristics of digital models.  There's two ways of doing this: denser models or models with more integration points per element.  Here's a good reference:
https://deust.wordpress.com/2014/11/30/h-method-p-method/#:~:text=This%20method%20refers%20to%20increasing,method%20uses%20few%20complex%20elements.

The cool part for us is we can visualize the geometric curvature inherent in "higher order" elements when the points along the edges are not placed directly between their corresponding edge nodes.

Reply:
Looks like a good reference, thanks! Hopefully the link won't be too mobile in the future.

Reply:
Also thanks for the explanation (it didn't load before I left my previous comment). I had very limited exposure to finite elements and I haven't heard about nonlinear elements. They indeed look useful :)